### PR TITLE
Don't use .result() in BigQueryOfflineStore, since it still leads to OOM

### PR DESCRIPTION
**Pull request by tsotnet**
Signed-off-by: Tsotne Tabidze <tsotne@tecton.ai>

**What this PR does / why we need it**: Follow up on #1638, which failed to fix the OOM issue.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/feast/infra/offline_stores/bigquery.py [hunk 1]
```diff
@@ -91,18 +92,28 @@ def get_historical_features(
...
 
         if type(entity_df) is str:
-            entity_df_job = client.query(entity_df)
-            # Start the job and get the schema back. We don't need the actual rows.
-            entity_df_result = entity_df_job.result(max_results=0)
+            entity_df_job: QueryJob = client.query(entity_df)
+            # Note: Don't call entity_df_job.result() here, it will lead
+            # to OOM errors. Instead just wait until the job is done.
+            while not entity_df_job.done():
+                time.sleep(1)
```
Comment (woop) [inline sdk/python/feast/infra/offline_stores/bigquery.py:99]: `done()` has a timeout and retry btw.

File: sdk/python/feast/infra/offline_stores/bigquery.py [hunk 2]
```diff
@@ -91,18 +92,28 @@ def get_historical_features(
...
 
+            entity_df_sql_table = f"`{entity_df_job.destination.project}.{entity_df_job.destination.dataset_id}.{entity_df_job.destination.table_id}`"
+
+            # Get a single row from this entity table to validate its schema
+            entity_df_limited_job: QueryJob = client.query(
```
Comment (woop) [inline sdk/python/feast/infra/offline_stores/bigquery.py:104]: This seems like a lighter weight approach https://github.com/feast-dev/feast/pull/1495#discussion_r619486530

File: sdk/python/feast/infra/offline_stores/bigquery.py (hunk 1 above)
Comment (tsotnet) [reply sdk/python/feast/infra/offline_stores/bigquery.py]: Didn't realize that, but with the new approach won't be using done(), so I'm resolving this.

File: sdk/python/feast/infra/offline_stores/bigquery.py (hunk 2 above)
Comment (tsotnet) [reply sdk/python/feast/infra/offline_stores/bigquery.py]: I'm trying this one out now. The previous approach still didn't work for super large datasets. Looks like the anonymous tables were not created when the dataset is large. I'll try creating named table instead.

**Review by woop — APPROVED** [review]

**Pull request merged**